enumerables.rb

```diff
@@ -0,0 +1,155 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  # Task-1 my_each exercise
+  def my_each
+    if block_given?
+      length.times { |e| yield(self[e]) }
+    else
+      to_enumerator(:my_each)
+    end
+  end
+
+  # Test for my_each
+  # [1,2,3].my_each {|e| p "#{e}"}
+  # Task-2 my_each_with_index
+  def my_each_with_index
+    if block_given?
+      (0...length).each { |index| yield(self[index], index) }
+    else
+      to_enumerator(:my_each_with_index)
+    end
+  end
```

Comment:
#my_each_with_index returns an enumerator if no block is given
#my_each_with_index when a block is given when self is an array returns the array itself after 
calling the given block once for each element in self
#my_each_with_index when a block is given when self is a range returns the range itself after 
calling the given block once for each element in self
#my_each_with_index when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self
#my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self